In [1]:
%pylab inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
from astropy.table import Table
from astropy.coordinates import SkyCoord
import astropy.units as u

In [3]:
gal= Table.read('BOSSDAT.fits')
quasars= Table.read('qso.fits')

In [4]:
print gal, quasars

PLATE  MJD  FIBER         Z                RA            DEC        OBJC_TYPE
----- ----- ----- ------------------ ------------- ---------------- ---------
 3586 55181     1     0.465112239122 9.33191224076  -0.462954655955         3
 3586 55181     2     0.719726264477 9.33007790809  -0.624115895012         3
 3586 55181     3      0.58804833889 9.43210592532  -0.245181536013         3
 3586 55181     5     0.585018992424 9.36867467866  -0.339224012425         3
 3586 55181     6     0.532988071442 9.40915664424  -0.250057843342         3
 3586 55181     7     0.474424600601 9.36047036958  -0.212842368574         3
 3586 55181     8     0.489556133747 9.35193235597  -0.159622251105         3
 3586 55181     9     0.489935219288 9.45202747482  -0.105271414218         3
 3586 55181    10     0.464580029249 9.47772265588  -0.131829129093         3
 3586 55181    11     0.490401208401 9.47710888524 -0.0977252219324         3
  ...   ...   ...                ...           ...              

In [5]:
def dist(i,j):
    qa = SkyCoord(quasars['RA'][i], quasars['DEC'][i], "icrs", unit="deg")
    ga = SkyCoord(gal['RA'][j], gal['DEC'][j], "icrs", unit="deg") 
    return qa.separation(ga).to(u.arcsec).value

In [6]:
qatalog = SkyCoord(quasars['RA']*u.degree, quasars['DEC']*u.degree ) 
gatalog = SkyCoord(gal['RA']*u.degree, gal['DEC']*u.degree ) 

In [7]:
idx, d2d, d3d = qatalog.match_to_catalog_sky(gatalog)

In [15]:
closest = np.argsort(d2d)  
#argsort - sorts out
#d2d - The on-sky separation between the closest match for each element in this object
#idk - 

In [24]:
for k in range(0,3000,2):
    i=closest[k]
        #i=closest[k] - gets every other one so doesnt repeat
    j=idx[i]
        #idx - gets the pair of i
    if quasars['RA'][i] != gal['RA'][j] and quasars['DEC'][i] != gal['DEC'][j]:
        print i, j, dist(i,j), gal['PLATE'][j], gal['MJD'][j], gal['FIBER'][j], quasars['PLATE'][i], quasars['MJD'][i], quasars['FIBER'][i]

69061 1216901 0.0123149981576 6398 56356 898 7515 56781 292
112238 1336615 2.02901631567 6659 56607 968 6656 56624 336
172579 1340827 2.43263115148 6666 56371 713 7416 56810 552
107193 450344 2.48716436514 4563 55617 100 4559 55597 482
63709 733457 2.70737935553 5163 55889 893 7326 56710 11
175237 1340923 2.75845839133 6666 56371 873 6665 56390 529
89480 1530356 2.81973848458 7299 56686 577 7298 56661 43
241940 1378028 3.00669097972 6728 56426 109 6729 56419 516
241730 626071 3.06344370029 4892 55738 327 4891 55736 542
137673 1359518 3.09323559564 6698 56637 129 6691 56413 822
165496 1488728 3.25354580049 7118 56686 252 6968 56443 824
208575 1415472 3.31611708217 6808 56443 588 6815 56419 162
134789 1355867 3.3665241954 6692 56390 71 6685 56412 522
16067 498548 3.53808713781 4660 56191 413 5706 56165 914
86516 929263 3.57465438692 5766 56248 74 5783 56017 514
170459 1340724 3.62145092271 6666 56371 548 7419 56811 884
17022 308301 3.63847162734 4311 55506 288 4225 55455 901
162773 10240

print quasars['PLATE'][69061], quasars['MJD'][69061], quasars['FIBER'][69061], gal['PLATE'][1216901], gal['MJD'][1216901], gal['FIBER'][1216901]

In [63]:
print quasars['PLATE'][34310], quasars['MJD'][34310], quasars['FIBER'][34310], gal['PLATE'][1518858], gal['MJD'][1518858], gal['FIBER'][1518858]

7236 56605 348 7236 56605 348
